src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -51,13 +58,17 @@ def resolve_requirements(self, interpreter, req_libs):
       else:
         target_set_id = 'no_targets'
 
+      # We need to ensure that we are resolving for only the current platform if we are
+      # including local python dist targets that have native extensions.
+      maybe_platforms = ['current'] if self.tgt_closure_has_native_sources else None
```

Comment:
It's probably better to only set maybe_platforms to `current` if the default configuration has more than one platform and current is in the platform list.